src/main/java/EmptyCollectionException.java

```diff
@@ -0,0 +1,5 @@
+public class EmptyCollectionException extends RuntimeException {
+    public EmptyCollectionException(String message) {
+        super(message);
+    }
+}
```

Comment:
Exceptions should always have the four public constructors, with/without message and cause. See the public constructors of [RuntimeException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/RuntimeException.html). There's cases where you wouldn't, but they're pretty few and far between.